preload.sh

```diff
@@ -13,76 +13,188 @@ API_HOST=${API_HOST:-"https://api.resin.io"}
 REGISTRY_HOST=${REGISTRY_HOST:-"registry.resin.io"}
 
 function cleanup {
-        rm $TMP_APPS_JSON || true
-	test "$DOCKER_PID" && kill $(cat $DOCKER_PID)
-	echo "Waiting for Docker to stop..."
-	while [ -e "$DOCKER_PID" ]; do
-		sleep 1
-	done
-        test -d "/tmp/docker-$APP_ID" && rm -rf "/tmp/docker-$APP_ID"
-        test "`mount | grep \"/mnt/$APP_ID\"`" && umount "/mnt/$APP_ID"
-        test -d "/mnt/$APP_ID" && rmdir "/mnt/$APP_ID"
+  test -e $TMP_APPS_JSON && rm $TMP_APPS_JSON || true
+  test "$DOCKER_PID" && kill $(cat $DOCKER_PID)
+  
+  echo "Waiting for Docker to stop..."
+  while [ -e "$DOCKER_PID" ]; do
+    sleep 1
+  done
+  
+  test -d "/tmp/docker-$APP_ID" && rm -rf "/tmp/docker-$APP_ID"
+  test "`mount | grep \"/mnt/$APP_ID\"`" && umount "/mnt/$APP_ID" || true
+  test -d "/mnt/$APP_ID" && rmdir "/mnt/$APP_ID"
+  
+  # Detach all /dev/loop interfaces
+  losetup -D
 }
 
 trap cleanup EXIT
 
-# Get app data and write to temporary file
-
-TMP_APPS_JSON=$(mktemp)
-
-( if test "$API_TOKEN"; then
-        curl -sH "Authorization: Bearer $API_TOKEN" "$API_HOST/ewa/application($APP_ID)?\$expand=environment_variable"
-elif test "$API_KEY"; then
-        curl "$API_HOST/ewa/application($APP_ID)?\$expand=environment_variable&apikey=$API_KEY"
-fi ) | jq --arg registryHost "$REGISTRY_HOST" '.d[0] |
-        (.git_repository | split("/") | .[1] | rtrimstr(".git")) as $repoName |
-        ($repoName + "/" + .commit) as $imageRepo |
-        ($registryHost + "/" + $imageRepo) as $imageId |
-	((.environment_variable // []) | map({(.name): .value}) | add) as $env |
-        [ { appId: .id, commit, imageRepo: $imageRepo, imageId: $imageId, env: $env } ]' > "$TMP_APPS_JSON"
-
-IMAGE_REPO=$(jq -r '.[0].imageRepo' "$TMP_APPS_JSON")
-echo "Loading the following image: " $IMAGE_REPO
-
-# Get application container size
-
-IMAGE_ID=$(curl -s "$REGISTRY_HOST/v1/repositories/$IMAGE_REPO/tags/latest" | jq -r '.')
-
-CONTAINER_SIZE=$(curl -s "$REGISTRY_HOST/v1/images/$IMAGE_ID/ancestry" | \
-jq '.[]' | awk '{print "'$REGISTRY_HOST'/v1/images/" $1 "/json"}' | \
-xargs -r -n 1 curl -I -s | \
-grep 'X-Docker-Size' | \
-awk '{s+=$2} END {print int(s / 1000000)}')
-echo "container size: " $CONTAINER_SIZE "MB"
+function version() {
+  echo "$@" | sed 's/^(\d+(\.\d+)*)/\1/' | awk -F. '{ printf("%03d%03d%03d\n", $1,$2,$3); }';
+}
 
-# Size will be increased by 110% of container size
-IMG_ADD_SPACE=$(expr $CONTAINER_SIZE / 100 + 300)
+# Compare two version numbers
+# Usage:
+# if version_ge $first_version $second_version; then
+#   echo "$first_version >= $second_version !"
+# fi
+function version_ge() {
+  [[ $(version $1) > $(version $2) || $(version $1) = $(version $2) ]]
```

Comment:
This can be
```bash
[[ $(version $1) -ge $(version $2) ]]
```
which also performs an integer comparison rather than string comparison